### PR TITLE
test images: Fixes echoserver s390x image

### LINKED ISSUE #84158: tests: Kubernetes E2E test image echoserver no longer builds
**What happened**:

The Kubernetes E2E image ``echoserver`` no longer builds. This error is encountered when building:

    Building image for echoserver ARCH: s390x...
    ~/kubernetes/_tmp/test-images-build.hI1x5F ~/kubernetes/test/images
    Registering qemu-*-static binaries in the kernel
    Setting /usr/bin/qemu-alpha-static as binfmt interpreter for alpha
    Setting /usr/bin/qemu-arm-static as binfmt interpreter for arm
    Setting /usr/bin/qemu-armeb-static as binfmt interpreter for armeb
    Setting /usr/bin/qemu-sparc32plus-static as binfmt interpreter for sparc32plus
    Setting /usr/bin/qemu-ppc-static as binfmt interpreter for ppc
    Setting /usr/bin/qemu-ppc64-static as binfmt interpreter for ppc64
    Setting /usr/bin/qemu-ppc64le-static as binfmt interpreter for ppc64le
    Setting /usr/bin/qemu-m68k-static as binfmt interpreter for m68k
    Setting /usr/bin/qemu-mips-static as binfmt interpreter for mips
    Setting /usr/bin/qemu-mipsel-static as binfmt interpreter for mipsel
    Setting /usr/bin/qemu-mipsn32-static as binfmt interpreter for mipsn32
    Setting /usr/bin/qemu-mipsn32el-static as binfmt interpreter for mipsn32el
    Setting /usr/bin/qemu-mips64-static as binfmt interpreter for mips64
    Setting /usr/bin/qemu-mips64el-static as binfmt interpreter for mips64el
    Setting /usr/bin/qemu-sh4-static as binfmt interpreter for sh4
    Setting /usr/bin/qemu-sh4eb-static as binfmt interpreter for sh4eb
    Setting /usr/bin/qemu-s390x-static as binfmt interpreter for s390x
    Setting /usr/bin/qemu-aarch64-static as binfmt interpreter for aarch64
    Setting /usr/bin/qemu-aarch64_be-static as binfmt interpreter for aarch64_be
    Setting /usr/bin/qemu-hppa-static as binfmt interpreter for hppa
    Setting /usr/bin/qemu-riscv32-static as binfmt interpreter for riscv32
    Setting /usr/bin/qemu-riscv64-static as binfmt interpreter for riscv64
    Setting /usr/bin/qemu-xtensa-static as binfmt interpreter for xtensa
    Setting /usr/bin/qemu-xtensaeb-static as binfmt interpreter for xtensaeb
    Setting /usr/bin/qemu-microblaze-static as binfmt interpreter for microblaze
    Setting /usr/bin/qemu-microblazeel-static as binfmt interpreter for microblazeel
    Sending build context to Docker daemon  3.256MB
    Step 1/10 : FROM s390x/nginx:1.15-alpine
    1.15-alpine: Pulling from s390x/nginx
    Digest: sha256:ca07edf3336d3ed6511687852e8d139166c74ddcbe642574799a12d8afc97477
    Status: Image is up to date for s390x/nginx:1.15-alpine
     ---> c1dc44096563
    Step 2/10 : COPY qemu-s390x-static /usr/bin/
     ---> Using cache
     ---> 78d3c3541ee2
    Step 3/10 : RUN apk add --no-cache openssl nginx-mod-http-lua nginx-mod-http-lua-upstream
     ---> Running in 8b95bb8fe0e0
    ERROR: Unable to open root: Bad address
    ERROR: Failed to open apk database: Bad address
    The command '/bin/sh -c apk add --no-cache openssl nginx-mod-http-lua nginx-mod-http-lua-upstream' returned a non-zero code: 99
    Makefile:37: recipe for target 'all-container' failed
    make: *** [all-container] Error 99

This only occurs for the ``s390x`` image.

**What you expected to happen**:

``echoserver`` should be buildable.

**How to reproduce it (as minimally and precisely as possible)**:

    cd kubernetes/test/images
    make all-push WHAT=echoserver

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g: `cat /etc/os-release`): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Network plugin and version (if this is a network-related bug): N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind bug
/sig testing
/area conformance

**What this PR does / why we need it**:

When trying to build the s390x image, it would fail when running the apk command with the following error:

    ERROR: Unable to open root: Bad address
    ERROR: Failed to open apk database: Bad address

This can be fixed by updating the ``third_party/multiarch/qemu-user-static/register/register.sh`` and ``third_party/multiarch/qemu-user-static/register/qemu-binfmt-conf.sh`` scripts and their usage to a newer version [1].

Additionally, the packages ``nginx-mod-http-lua`` and ``nginx-mod-http-lua-upstream`` cannot be found in the regular ``http://dl-cdn.alpinelinux.org/alpine/v3.9/main/s390x/`` repository, but we can use an older one [2].

[1] https://github.com/qemu/qemu/blob/master/scripts/qemu-binfmt-conf.sh
[2] http://dl-cdn.alpinelinux.org/alpine/v3.8/main

**Which issue(s) this PR fixes**:

Fixes #84158

**Special notes for your reviewer**:

Image building logs with the current changes: https://paste.ubuntu.com/p/zrwVfbbS75/

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/images/echoserver/install-deps.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# limitations under the License.
+
+ARCH=$(uname -m)
+
+if [ "${ARCH}" = "s390x" ]; then
+    echo "http://dl-cdn.alpinelinux.org/alpine/v3.8/main" > /etc/apk/repositories
+    echo "http://dl-cdn.alpinelinux.org/alpine/v3.8/community" >> /etc/apk/repositories
+fi
+
+apk add -f openssl nginx-mod-http-lua nginx-mod-http-lua-upstream
```
Comment (mkumatag) [inline test/images/echoserver/install-deps.sh:24]: --no-cache option is missing here..

**Review by mkumatag — CHANGES_REQUESTED** [review]
Rest I assume that we just copied the updated scripts..

File: test/images/echoserver/install-deps.sh (hunk 1 above)
Comment (claudiubelu) [reply test/images/echoserver/install-deps.sh:24]: Missed that. Added.

**Pull request merged**